tests/unit/test_adapter.py

```diff
@@ -105,8 +105,60 @@ def test_measurements_adapter(iJO1366):
             "uncertainty": 0,
         },
     ]
+    # Made up proteomics data:
+    proteomics = [
+        {
+            "identifier": "P0A8V2",  # protein not in model (should be skipped)
+            "measurement": 5.03e-6,
+            "uncertainty": 0,
+        },
+        {
+            "identifier": "P0AFG8",
+            "measurement": 8.2e-3,  # very high value (should be kept)
+            "uncertainty": 8.2e-6,
+        },
+        {
+            "identifier": "P15254",
+            "measurement": 6.54e-8,  # very low value (should be removed)
+            "uncertainty": 0,
+        },
+        {
+            "identifier": "P0A6C5",
+            "measurement": 5.93e-8,  # very low value (should be removed)
+            "uncertainty": 0,
+        },
+    ]
     operations, warnings, errors = apply_measurements(
-        iJO1366, biomass_reaction, fluxomics, [], uptake_secretion_rates, [], None
+        iJO1366,
+        biomass_reaction,
+        is_ec_model,
+        fluxomics,
+        [],
+        proteomics,
+        uptake_secretion_rates,
+        [],
+        None,
     )
+    # 4 operations (2 rates + 2 fluxomics) + 1 warning (not an ecModel) are expected:
     assert len(operations) == 4
+    assert len(warnings) == 1
+    assert len(errors) == 0
+
+    eciML1515, biomass_reaction, is_ec_model = eciML1515
+    growth_rate = {"measurement": 0.1, "uncertainty": 0.01}
+    operations, warnings, errors = apply_measurements(
+        eciML1515,
+        biomass_reaction,
+        is_ec_model,
+        [],
+        [],
+        proteomics,
+        [],
+        [],
+        growth_rate,
+    )
+    # 2 operations (1 proteomics + growth rate) + 1 warning (1 skipped protein) are
+    # expected:
+    assert len(operations) == 2
+    assert len(warnings) == 1
```

Comment:
To keep the tests as small and encapsulated as possible, it could make sense to split it up and move this part to a separate test.

Reply:
I wrote it like this bc in that section we are overall testing the same function (`test_measurements_adapter`), but if you think it's preferable to split will do :)

Reply:
update: fixed in [`074f6f6`](https://github.com/DD-DeCaF/simulations/pull/147/commits/074f6f641120df7e6aecba44a4fcb2c4ddafae72)